HDFS-12292, HDFS-11432, HDFS-7600, HDFS-11177, HDFS-11011. Support viewfs:// schema path for DfsAdmin commands

MikhailErofeev opened the pull request:
HDFS-12292, HDFS-11432, HDFS-7600, HDFS-11177, HDFS-11011. Support viewfs:// schema path for DfsAdmin commands

HDFS-12292 Federation: Support viewfs:// schema path for DfsAdmin commands. Contributed by Mikhail Erofeev.

(not in upstream)

HDFS-11432. Federation : Support fully qualified path for Quota/Snapshot/cacheadmin/cryptoadmin commands. Contributed by Brahma Reddy Battula.

(cherry picked from commit dcd03df)

HDFS-7600. Refine hdfs admin classes to reuse common code. Contributed by Jing Zhao.

(cherry picked from commit 6f3a63a)

HDFS-11177. 'storagepolicies -getStoragePolicy' command should accept URI based path. (Contributed by Surendra Singh Lilhore)

(cherry picked from commit 4804050)

HDFS-11011. Add unit tests for HDFS command 'dfsadmin -set/clrSpaceQuota'. Contributed by Xiaobing Zhou.

(cherry picked from commit 9a8a386)

MikhailErofeev commented:
No manual tests yet.

ashangit commented:
Patch looks good.
Can you update the head of the commit message so all request on which it rely are viewable on it.
It will be more easy to fastly see patches that we should take in account if we migrate to new cdh release

Commit:
HDFS-11432, HDFS-7600, HDFS-11177, HDFS-11011. Federation : Support fully qualified path for Quota/Snapshot/cacheadmin/cryptoadmin commands.

HDFS-11432. Federation : Support fully qualified path for Quota/Snapshot/cacheadmin/cryptoadmin commands. Contributed by Brahma Reddy Battula.

(cherry picked from commit dcd03df)

HDFS-7600. Refine hdfs admin classes to reuse common code. Contributed by Jing Zhao.

(cherry picked from commit 6f3a63a)

HDFS-11177. 'storagepolicies -getStoragePolicy' command should accept URI based path. (Contributed by Surendra Singh Lilhore)

(cherry picked from commit 4804050)

HDFS-11011. Add unit tests for HDFS command 'dfsadmin -set/clrSpaceQuota'. Contributed by Xiaobing Zhou.

(cherry picked from commit 9a8a386)

MikhailErofeev commented:
The patch has been rebased with https://issues.apache.org/jira/browse/HDFS-12292 v.4 and a proper comment

Commit:
HDFS-12292. Federation: Support viewfs:// schema path for DfsAdmin commands. Contributed by Mikhail Erofeev.

(not in upstream)

ashangit commented on hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/PathData.java:
```diff
@@ -308,6 +308,30 @@ private String getStringForChildPath(Path childPath) {
+    }
+    // FileStatus contains unresolved path thus should be replaced too
+    FileStatus resolvedStat = resolvedFs.getFileStatus(resolvedPath);
+    return new PathData(resolvedFs, resolvedPath.toString(), resolvedStat);
```
Perhaps you can use return new PathData(resolvedPath.toString(), conf); to simplify the code

ashangit replied:
or return new PathData(resolvedFs, resolvedPath.toString());

ashangit commented:
One OPTS comment otherwise LGTM
